main.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/status-im/msgstat/stats"
+)
+
+var (
+	gitCommit    = ""      // rely on linker: -ldflags -X main.gitCommit"
+	buildStamp   = ""      // rely on linker: -ldflags -X main.buildStamp"
+	versionStamp = "0.0.1" // rely on linker -ldflags -X main.versionStamp"
+)
+
+var (
+	sourceFile = flag.String("file", "", "input file containing status-go logs")
+	targetFile = flag.String("out", "", "output file to save processed data into")
+	format     = flag.String("format", "json", "format of output data. -format=json|yaml|csv")
+	version    = flag.Bool("v", false, "Print version")
+)
+
+type wopCloser struct {
+	io.Writer
+}
+
+// Close does nothing.
+func (wopCloser) Close() error {
+	return nil
+}
+
+func main() {
```

Comment:
As usual I prefer an extra `Main` type and in `main()` only create a config (here flags, sometimes reading configuration files or environment variables), then create the `Main` instance with the configuration and then simply call `Run()` on this type.

Reason is a better unit testing of this `Main` type.

Reply:
Considering we have test in `stats` subpkg that validate the behaviour we want currently, am not sure testing an extra `Main` would be needed. Unless we wish to validate we can create files and write to them since `stats` already has an associated test in https://github.com/status-im/msgstat/blob/feature/stats-aggregation/stats/stats_test.go.